docs/using.rst

```diff
@@ -653,8 +653,8 @@ Example usage for DNS-01 (Cloudflare API v4) (for example purposes only, do not
    API_KEY="your-api-key"
    EMAIL="your.email@example.com"
 
-   # Strip only the top domain to get the zone id
-   DOMAIN=$(expr match "$CERTBOT_DOMAIN" '.*\.\(.*\..*\)')
+   # Fetch the SOA domain for the domain/subdomain
+   DOMAIN=$(dig soa "$CERTBOT_DOMAIN" | grep -v ^\; | grep SOA | awk '{print $1}';)
```

Comment:
We are adding another dependency (even though curl was one already) here, so maybe a note should be added telling the user that dig should be made available. I don't think it's included in many default installations.